sw/device/tests/sram_ctrl_execution_test_ret.c

```diff
@@ -0,0 +1,178 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#include "sw/device/lib/base/csr.h"
+#include "sw/device/lib/base/macros.h"
+#include "sw/device/lib/base/stdasm.h"
+#include "sw/device/lib/dif/dif_sram_ctrl.h"
+#include "sw/device/lib/handler.h"
+#include "sw/device/lib/runtime/ibex.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/testing/check.h"
+#include "sw/device/lib/testing/sram_ctrl_testutils.h"
+#include "sw/device/lib/testing/test_framework/test_main.h"
+#include "sw/device/lib/testing/test_framework/test_status.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+
+const test_config_t kTestConfig;
+
+static dif_sram_ctrl_t sram_ctrl;
+
+/**
+ * This flag is used to verify that the instruction access fault has occurred,
+ * and has been successfully serviced. Declared as volatile, because it is
+ * referenced in the fault handler, as well as the main test flow.
+ */
+static volatile bool instruction_access_fault;
+
+static const uint32_t kRetRamStartAddr =
+    TOP_EARLGREY_SRAM_CTRL_RET_AON_RAM_BASE_ADDR;
+static const uint32_t kRetRamEndAddr =
+    TOP_EARLGREY_SRAM_CTRL_RET_AON_RAM_BASE_ADDR +
+    TOP_EARLGREY_SRAM_CTRL_RET_AON_RAM_SIZE_BYTES - 1;
+
+/**
+ * Buffer filled with `0` (valid `unimp`) instructions.
+ *
+ * When Retention SRAM is disabled, executing this buffer will result in an
+ * Instruction Access Fault, which constitutes a successful test. Otherwise,
+ * it will result in an Illegal Instruction Fault, which constitutes a failure.
+ */
+static const uint32_t execution_test_instructions[2];
```

Comment:
If we're going with this approach then we probably don't need this variable. We can just zero out the target buffer directly (though do copy the comment across).